staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -174,117 +184,171 @@ func NewCmdApply(baseName string, f cmdutil.Factory, ioStreams genericclioptions
-		return fmt.Errorf("--dry-run=client doesn't work with --server-side (did you mean --dry-run=server instead?)")
-	}
+	dryRunVerifier := resource.NewDryRunVerifier(dynamicClient, flags.Factory.OpenAPIGetter())
+	fieldManager := GetApplyFieldManagerFlag(cmd, serverSideApply)
 
 	// allow for a success message operation to be specified at print time
-	o.ToPrinter = func(operation string) (printers.ResourcePrinter, error) {
-		o.PrintFlags.NamePrintFlags.Operation = operation
-		cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
-		return o.PrintFlags.ToPrinter()
+	toPrinter := func(operation string) (printers.ResourcePrinter, error) {
+		flags.PrintFlags.NamePrintFlags.Operation = operation
+		cmdutil.PrintFlagsWithDryRunStrategy(flags.PrintFlags, dryRunStrategy)
+		return flags.PrintFlags.ToPrinter()
 	}
 
-	o.RecordFlags.Complete(cmd)
-	o.Recorder, err = o.RecordFlags.ToRecorder()
+	flags.RecordFlags.Complete(cmd)
+	recorder, err := flags.RecordFlags.ToRecorder()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	o.DeleteOptions, err = o.DeleteFlags.ToOptions(o.DynamicClient, o.IOStreams)
+	deleteOptions, err := flags.DeleteFlags.ToOptions(dynamicClient, flags.IOStreams)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	err = o.DeleteOptions.FilenameOptions.RequireFilenameOrKustomize()
+	err = deleteOptions.FilenameOptions.RequireFilenameOrKustomize()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if o.ServerSideApply && o.DeleteOptions.ForceDeletion {
-		return fmt.Errorf("--force cannot be used with --server-side")
-	}
-
-	if o.DryRunStrategy == cmdutil.DryRunServer && o.DeleteOptions.ForceDeletion {
-		return fmt.Errorf("--dry-run=server cannot be used with --force")
-	}
-
-	o.OpenAPISchema, _ = f.OpenAPISchema()
-	o.Validator, err = f.Validator(cmdutil.GetFlagBool(cmd, "validate"))
+	openAPISchema, _ := flags.Factory.OpenAPISchema()
+	validator, err := flags.Factory.Validator(cmdutil.GetFlagBool(cmd, "validate"))
 	if err != nil {
-		return err
+		return nil, err
 	}
-	o.Builder = f.NewBuilder()
-	o.Mapper, err = f.ToRESTMapper()
+	builder := flags.Factory.NewBuilder()
+	mapper, err := flags.Factory.ToRESTMapper()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	o.Namespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	namespace, enforceNamespace, err := flags.Factory.ToRawKubeConfigLoader().Namespace()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if o.Prune {
-		o.PruneResources, err = parsePruneResources(o.Mapper, o.PruneWhitelist)
+	if flags.Prune {
+		flags.PruneResources, err = parsePruneResources(mapper, flags.PruneWhitelist)
 		if err != nil {
-			return err
+			return nil, err
 		}
 	}
 
+	o := &ApplyOptions{
+		// 	Store baseName for use in printing warnings / messages involving the base command name.
+		// 	This is useful for downstream command that wrap this one.
+		cmdBaseName: baseName,
```

Comment:
Get rid of that, it's useless, old pattern.

Reply:
Do you want me to replace its occurance in the logs with hardcoded value "kubectl"?

Log - ```fmt.Fprintf(o.ErrOut, warningNoLastAppliedConfigAnnotation, info.ObjectName(), corev1.LastAppliedConfigAnnotation, o.cmdBaseName)```